core/platform/email/mailgun_email_services.py

```diff
@@ -120,7 +126,7 @@ def send_email_to_recipients(
         recipient_emails[i:i + 1000]
         for i in range(0, len(recipient_emails), 1000)]
     for email_list in recipient_email_lists:
-        data = {
+        data: Dict[str, Any] = {
```

Comment:
Referred https://stackoverflow.com/a/68807635 to fix the typing error by explicitly declaring type. However, it's unclear as to why this was only captured after I made the mailgun domain name change.

@seanlip let me know if I should file an issue for this.